### DISCUSSION TITLE
在使用vc-imagery-provider-tianditu控制台报如下错误

### DISCUSSION THREAD

**Discussion by eliduty** [zouyaoji/vue-cesium#335] [Q&A] [answered]
在使用天地图影像图时，有些时候可以渲染出来，有些时候渲染不出来，会出现如下图的错误，存在及一定的机率。
![image](https://user-images.githubusercontent.com/13143017/209813345-650acad3-f0d2-4d3c-83d9-449a37e404a0.png)
当换其他影像图时正常，测试过vc-imagery-provider-arcgis，想请问一下是什么问题，已经困扰一段时间了。

cesiumPath：使用的VcConfigProvider的默认值，也测试过本地的Cesium.js 问题一致
"vue-cesium": "3.1.2" 版本

**Comment by zouyaoji** [comment]
未能够复现，请提供复现问题的代码哈。

**Reply by eliduty** [reply]
![image](https://user-images.githubusercontent.com/13143017/210032778-39479a73-1001-4daf-9ec0-6345734b2d9f.png)
![image](https://user-images.githubusercontent.com/13143017/210032807-4ee39645-2374-43a6-888c-53f11ef0828e.png)
使用注释掉的vc-imagery-provider-arcgis是正常的，我这边也不是每次会出现这个问题，我初步怀疑是vc-imagery-provider-tianditu中未获取到Cesium导致的，可能和加载的先后顺序有关。
目前我的解决方案是监听vc-viewer加载成功后再渲染vc-layer-imagery这样就不会出现上述的错误。
代码中注释掉的isReady就是在cesiumReady事件响应后进行的赋值。

**Reply by zouyaoji** [reply]
vc-imagery-provider-arcgis 和 vc-imagery-provider-tianditu 分别用 vc-layer-imagery包裹哈。

**Reply by eliduty** [reply]
没有同时启用vc-imagery-provider-arcgis 和 vc-imagery-provider-tianditu。

**Reply by zouyaoji** [reply, marked as answer]
提问者已经自己找到解决方案
如果vc-viewer所在路由有频繁切换，把vc-viewer 的removeCesiumScript设置成了false就对了
